### ISSUE TITLE
Should we propagate causes in AbortedExecutionException?

### ISSUE THREAD

**Issue by slinkydeveloper** [restatedev/sdk-java#157]
Please see #156 first.

## Status quo

With https://github.com/restatedev/sdk-java/pull/153 we (hopefully) fixed all the various non-deterministic exceptions that can pop out of `RestateContext` methods, such that the following rule applies:

```java
 * <p>All methods of this interface, and related interfaces, throws either {@link TerminalException}
 * or {@link AbortedExecutionException}, where the former can be caught and acted upon, while the
 * latter MUST NOT be caught, but simply propagated for clean up purposes.
```

This exception is used to unblock the user thread when suspending (we used to call it `SuspendedException` in past) and for failures that we don't record in the journal (such as non deterministic replay error, as for side effect retryable errors). For example:

```java
try {
    ctx.sideEffect(() -> {
        throw new IllegalStateException();
    });
} catch (IllegalStateException e) {
    // This will never be executed
} catch (AbortedExecutionException e) {
    // This will be executed, but users MUST NOT write this catch arm!
}
```

Before throwing the `AbortedExecutionException`, the SDK internal state machine logs the error to screen, sends back a message to the runtime to close the stream **WITHOUT** recording the failure in the journal, and then fences off any subsequent usage of the `RestateContext`. In other words, if you write this code:

```java
try {
  // do stuff
} catch (Throwable e) {
  ctx.sideEffect() // Or any other RestateContext method
}
```

if the throwed exception is `AbortedExecutionException`, then that side effect won't be executed and will just throw again `AbortedExecutionException`.

This same exception is also used to unblock the user thread when suspending (we used to call it `SuspendedException` in past).

As it stands now, the `AbortedExecutionException` is implemented as singleton and will always have a null cause, to avoid users write non-deterministic control flow based on the cause.

## Question

Should we rather expose the cause of the `AbortedExecutionException`? If yes, for what purpose?

**Comment by slinkydeveloper** [comment]
I think there's agreement in not doing this now, so i'm gonna close it as wontfix. Reopen if you need it.
